test/lambda.test.js

```diff
@@ -1260,6 +1260,182 @@ test('[lambda] status: success', (assert) => {
+    .onCall(1).callsFake(() => Promise.reject({
+      statusCode: 404,
+      statusMessage: 'Not Found'
+    }));
+
+  sinon.stub(got, 'post')
+    .onCall(0).callsFake(() => Promise.resolve({
+      body: { token: 'v1.1f699f1069f60xxx' }
+    }))
+    .onCall(1).callsFake(() => Promise.reject({
+      statusCode: 422,
+      statusMessage: 'Unprocessable Entity'
+    }));
+
+  lambda.status(fakeStatusEvent, {}, (err) => {
+    const args = JSON.parse(JSON.stringify(got.post.args[0]));
+    const auth = args[1].headers.Authorization.replace('Bearer ', '');
+    const decoded = jwt.verify(auth, publicKey, { algorithms: ['RS256'] });
+
+    assert.ifError(err, 'does not error on missing GitSha');
+
+    assert.ok(
+      got.post.calledWith(
+        'https://api.github.com/repos/mapbox/stork/statuses/12345shanotfound',
+        {
+          json: true,
+          headers: {
+            'Content-type': 'application/json',
+            'User-Agent': 'github.com/mapbox/stork',
+            Accept: 'application/vnd.github.machine-man-preview+json',
+            Authorization: 'token v1.1f699f1069f60xxx'
+          },
+          body: JSON.stringify({
+            context: 'stork',
+            description: 'Your build succeeded',
+            state: 'success',
+            target_url: 'url for cloudwatch logs'
+          })
+        }
+      ),
+      'sets PR status via github api request'
+    );
+
+    assert.equal(got.get.callCount, 2, '2 get requests to github api');
+    assert.ok(
+      got.get.calledWith(
+        'https://api.github.com/repos/mapbox/stork/commits/12345shanotfound',
+        {
+          json: true,
+          headers: {
+            'User-Agent': 'github.com/mapbox/stork',
+            Accept: 'application/vnd.github.machine-man-preview+json',
+            Authorization: `Bearer ${auth}`
+          }
+        }
+      ),
+      'call to check if sha exists'
+    );
+
+    environment.restore();
+    got.get.restore();
+    got.post.restore();
+    lambda.decrypt.restore();
+    AWS.CodeBuild.restore();
+    assert.end();
+  });
+});
+
+test('[lambda] status: github error, sha exists', (assert) => {
+  const environment = env(statusVars).mock();
+
+  sinon.stub(lambda, 'decrypt').callsFake(fakeDecrypt);
+
+  const getBuild = AWS.stub('CodeBuild', 'batchGetBuilds', function() {
+    this.request.promise.returns(Promise.resolve({
+      builds: [
+        {
+          logs: { deepLink: 'url for cloudwatch logs' },
+          sourceVersion: '12345shaexists',
+          source: { location: 'https://github.com/mapbox/stork' }
+        }
+      ]
+    }));
+  });
+
+  sinon.stub(got, 'get')
+    .onCall(0).callsFake(() => Promise.resolve())
+    .onCall(1).callsFake(() => Promise.resolve({
+      statusCode: 200,
+      statusMessage: 'OK',
+      body: {
+        sha: '12345shaexists'
+      }
+    }));
+
+  sinon.stub(got, 'post')
+    .onCall(0).callsFake(() => Promise.resolve({
```

Comment:
@vsmart I just added the `post` call that results in the token, similar to what we did for the `gitsha not found` tests.